lib/ipmi_lanp.c

```diff
@@ -245,7 +245,7 @@ get_lan_param_select(struct ipmi_intf *intf, uint8_t chan, int param, int select
 			specific_val2str(rsp->ccode,
 			                 get_lan_cc_vals,
 			                 completion_code_vals));
-		return NULL;
+		return rc;
```

Comment:
In the commit log you claim that you return the pointer "in the case when the parameter is not supported", but it's not actually true as the `default` case covers all possible non-successful completion codes.
Moreover, the 'parameter is not supported' case is handled above in line 233.
Looks more like the above cases must not fall through to `default` and must print their own message, slightly different from what the `default` case must print.

Reply:
Note that in the default case `rc` will be NULL, and that we return `rc` as a way of returning NULL several times earlier in the function.

When looking at this issue, I realized that we currently set `rc` from this `case 0xcc` block but then we never use it, so I was thinking that my change here must have been the intended change that got missed somehow.

I do agree that's a bit tricky with the fall through, so I updated this case to add a different log message and return the pointer.

Reply:
Alexander, I think it's worth pointing out that the specific case in question (parameter not supported) was working fine up to commit 351dad24a26f56580ba6028a267c4924b7faa4cb. At this point the specific log message was removed and the fall-through to the default case was added. I think it was at that point where the code change was actually incomplete as the "rc = p" was added, while the default case has "return NULL" instead of "return p".

I believe that Daniel's original change fixed this oversight by simply changing the default to "return rc". 

In the latest version of the code "rc" is always NULL and cases of "return rc" should be replaced with "return NULL".

Can you please clarify whether 351dad24a26f56580ba6028a267c4924b7faa4cb changing the log message was intentional? If so, Daniel's latest change effectively undoes that.

I believe the commit log saying "you return the pointer "in the case when the parameter is not supported"" was correct, since in the default case rc = NULL, so in the default case nothing changed.